octopoes/connectors/services/xtdb.py

```diff
@@ -0,0 +1,171 @@
+"""XTDB HTTP client."""
+
+from __future__ import annotations
+
+import logging
+from datetime import timezone, datetime
+from enum import Enum
+from http import HTTPStatus
+from typing import Union, Optional, Tuple, List, Dict, Any, Type, Callable, cast
+
+import requests
+from pydantic import BaseModel, Field
+from requests import Response, HTTPError
+
+logger = logging.getLogger(__name__)
+
+
+class OperationType(Enum):
+    """Operation types for XTDB transactions."""
+
+    PUT = "put"
+    DELETE = "delete"
+    MATCH = "match"
+    EVICT = "evict"
+    FN = "fn"
+
+
+Operation = Tuple[OperationType, Union[str, Dict[str, Any]], Optional[datetime]]
+
+
+class Transaction(BaseModel):
+    """A XTDB transaction, consisting of a list of operations."""
+
+    operations: List[Operation] = Field(alias="tx-ops")
+
+    class Config:
+        """Serialize operations as 'tx-ops' in JSON."""
+
+        allow_population_by_field_name = True
+
+
+class XTDBHTTPSession(requests.Session):
```

Comment:
Although we've been doing it like this for a long time now; I think I would now prefer composition over inheritance since it currently doesn't add much. It's perfectly fine to just initialize a normal `request.Session` with a default header. It also doesn't make sense now in the `request` method, since it appends a `url` parameter to a base URL, whilst it is actually a path.

Maybe not really urgent, but just some thoughts I wanted to write down.

Reply:
Created a separate issue for this:
https://github.com/minvws/nl-kat-octopoes/issues/84